src/analysis/processing/qgsnativealgorithms.cpp

```diff
@@ -47,6 +47,7 @@
 #include "qgsalgorithmconstantraster.h"
 #include "qgsalgorithmconverttocurves.h"
 #include "qgsalgorithmconvexhull.h"
+#include "qgsalgorithmcoveragesimplify.h"
```

Comment:
Probably this should be put inside `ifdef` as algorithm requires GEOS >=3.12

Reply:
I think the "right" approach is that we still expose the algorithm and only catch the error when it's actually run. This means models shared with users with older geos won't be completely broken, but rather they'll get an explanatory error when they try to run them only.

But happy for second opinions!

Reply:
No, it is totally fine from this point of view.